### PR TITLE
Synchronize st2.conf with upstream

### PR DESCRIPTION
**Pull request by arm4b**
> Addresses https://github.com/StackStorm/chef-stackstorm/issues/25 and https://github.com/StackStorm/chef-stackstorm/issues/14

Official `st2.conf`: https://github.com/StackStorm/st2/blob/master/conf/st2.package.conf

### CODE REVIEW

File: recipes/_initial.rb [hunk 1]
```diff
@@ -25,5 +25,4 @@
   hostname 'localhost'
   comment 'Appended by st2 cookbook'
   action :append
```
Comment (arm4b) [inline recipes/_initial.rb]: Not sure about that entry if we need it at all. At least after using `127.0.0.1` instead of `localhost` this rule is out of scope of this cookbook.

From one point of view cleanup might be good, from another side: `127.0.0.1 localhost` in hosts won't hurt.

Ideas?

File: attributes/config.rb [hunk 2]
```diff
@@ -2,29 +2,38 @@
 
 # Config defaults options
 default['stackstorm']['config'] = {
-  api_url: 'http://localhost:9101',
-  conf_root: '/etc',
+  mask_secrets: true,
+  allow_origin: '*',
```
Comment (arm4b) [inline attributes/config.rb]: Thinking about renaming these config settings:
`allow_origin` => `api_allow_origin`
`mask_secrets` => `api_mask_secrets`

How it should be, according to: https://github.com/StackStorm/st2/blob/master/conf/st2.package.conf#L10

Let me know your opinions.

**Comment by arm4b** [comment]
BTW, related to this is to support _all_ `st2` config settings: https://github.com/StackStorm/chef-stackstorm/issues/35 what might be a good idea for real `prod` deployments

But it's not a thing for the nearest future.

**Comment by arm4b** [comment]
@jvrplmlmn @sysbot @shortdudey123 Please take a look and let's unblock this Cookbook.

Also, we got some more resources and would like to improve this repo.
This week plan: https://github.com/StackStorm/chef-stackstorm/milestone/1

I documented [More Features](https://github.com/StackStorm/chef-stackstorm/issues) recently. If there is something you would like to see here in future - create an Issue/Document it.

And thank you for your involvement!

File: spec/recipes/config_spec.rb [hunk 3]
```diff
@@ -120,31 +134,34 @@
         mode: 0644,
         source: 'st2.conf.erb',
         variables: {
```
Comment (shortdudey123) [inline spec/recipes/config_spec.rb:138]: we should alphabetize this

File: templates/default/st2.conf.erb [hunk 4]
```diff
@@ -3,30 +3,46 @@
...
 # allow_origin is required for handling CORS in st2 web UI.
 # allow_origin = http://myhost1.example.com:3000,http://myhost2.example.com:3000
 allow_origin = <%= @allow_origin %>
 
+[stream]
+logging = /etc/st2/logging.stream.conf
+
 [sensorcontainer]
-logging = <%= @conf_root %>/st2reactor/console.conf
+logging = /etc/st2/logging.sensorcontainer.conf
```
Comment (shortdudey123) [inline templates/default/st2.conf.erb:18]: why remove config option?

File: templates/default/st2.conf.erb [hunk 5]
```diff
@@ -3,30 +3,46 @@
 
 [api]
 # Host and port to bind the API server.
-host = 0.0.0.0
-port = 9101
-logging = <%= @conf_root %>/st2api/console.conf
+host = <%= @api_host %>
+port = <%= @api_port %>
+logging = /etc/st2/logging.api.conf
```
Comment (shortdudey123) [inline templates/default/st2.conf.erb:8]: why remove config option here as well?

**Review by shortdudey123 — CHANGES_REQUESTED** [review]

File: templates/default/st2.conf.erb (hunk 4 above)
Comment (arm4b) [reply templates/default/st2.conf.erb:18]: Yeah, included explanation in Commit message:

> You can't overwrite `conf_root` from `/etc/st2` to anything else, since it's globally hardcoded setting.

So `/etc/st2` is hardcoded in packages, you can't change it and it's used in so many places.
Allowing users to edit `conf_root` brings confusion and broken installation.

Comment (arm4b) [reply templates/default/st2.conf.erb:18]: It's outdated bit that is still here from the legacy installer, before introducing actual `.deb`/`.rpm` packages.

I'm going to touch more abandoned things like that in upcoming [Cookbook Cleanup](https://github.com/StackStorm/chef-stackstorm/issues/31).

File: spec/recipes/config_spec.rb (hunk 3 above)
Comment (arm4b) [reply spec/recipes/config_spec.rb:138]: I'm using it in a way as it appears in `st2.conf` for the reasons to verify synchronization easier in the future: 
https://github.com/StackStorm/st2/blob/master/conf/st2.dev.conf

File: recipes/_initial.rb (hunk 1 above)
Comment (shortdudey123) [reply recipes/_initial.rb]: agree

File: attributes/config.rb (hunk 2 above)
Comment (shortdudey123) [reply attributes/config.rb]: `api_` prepending makes more sense

File: spec/recipes/config_spec.rb (hunk 3 above)
Comment (shortdudey123) [reply spec/recipes/config_spec.rb:138]: If so, you should add a comment about that since whoever modifies this list next will just add it to the end since its not alphabetized

Comment (arm4b) [reply spec/recipes/config_spec.rb:138]: :+1: Good call!

Comment (shortdudey123) [reply spec/recipes/config_spec.rb:138]: much better :)

**Review by shortdudey123 — APPROVED** [review]

**Comment by shortdudey123** [comment]
LGTM for merging assuming tests pass
